### ISSUE TITLE
Error details: Error setting the binding dimension

### ISSUE THREAD

**Issue by Vinayaks117** [triton-inference-server/server#4798]
**Description**

I'm able to deploy fine-tuned "bert-base-uncased" model on Triton inference server using TensorRT while inference I am getting a following error.

Converted the onnx model to tensorrt using command below.
`trtexec --onnx=model.onnx --saveEngine=model_bs16.plan --minShapes=input_ids:1x128,attention_mask:1x128,token_type_ids:1x128 --optShapes=input_ids:16x128,attention_mask:16x128,token_type_ids:16x128 --maxShapes=input_ids:128x128,attention_mask:128x128,token_type_ids:128x128 --fp16 --verbose --workspace=14000 | tee conversion_bs16_dy.txt`

As per the above I am sending single batch at least but getting an error.

**Error details**
`E0822 12:50:12.125495 90 tensorrt.cc:1993] error setting the binding dimension`
`Error details: model expected the shape of dimension 1 to be between 128 and 128 but received 7`

**Triton Information**
Triton Server Version 2.22.0
NVIDIA Release 22.05

Using Triton container:
'007439368137.dkr.ecr.us-east-2.amazonaws.com/sagemaker-tritonserver:22.05-py3'

**Expected behavior**

Expected behavior
Basically, it receive a string (batch size = 1), send it to the tokenizer, get the output from the tokenizer and pass it as input of the model, return the model output.

I'm not able to get the inference due to above error. Kindly request you to help me on this issue.

Thanks a lot.

**Comment by rmccorm4** [comment]
Hi @Vinayaks117 ,

> Error details: model expected the shape of dimension 1 to be between 128 and 128 but received 7

It looks like your second dimension (dim 1) is fixed at size 128 according to your optimization profile shapes "Ax128, Bx128, Cx128, etc." for all inputs. 

Perhaps try setting the min shape to 1x1 or similar for the required inputs instead of 1x128 if you want to account for a smaller value such as 7. If the 7 for the second dimension is unexpected, then maybe there's an issue in forming the inference request/data.

**Comment by Vinayaks117** [comment]
Thanks for your reply @rmccorm4 

FYI: 7 --> second dimension (sequence length) is not unexpected because that is the shape of token_ids (1x7), sequence length is completely dynamic based on the number of tokens we have in inference text.

**Build model**
I see that onnx file input shapes have two dynamic axes: [batch, sequence] and built tensorrt model which accepts the batch size = 1 and max sequence length = 128.

`trtexec --onnx=model.onnx --saveEngine=model_default.plan --workspace=4096 --shapes=input_ids:1x128,attention_mask:1x128,token_type_ids:1x128`

![image](https://user-images.githubusercontent.com/37987800/186108377-ab1ed4b5-66fe-41c4-aab1-ad0b1b11ca09.png)

**Inference test**
When I load the engine then need to pass in the actual shape I would like to run the inference with by adding --shapes=input_ids:<N>x<P>,attention_mask:<N>x<P>,token_type_ids:<N>x<P>, where <N> is the actual batch size and <P> is the actual sequence length for inference.

**Default sequence length:**
`trtexec --loadEngine=model_default.plan --shapes=input_ids:1x128,attention_mask:1x128,token_type_ids:1x128`

![image](https://user-images.githubusercontent.com/37987800/186110083-fdb0f7a1-51ea-4f46-9f48-9a2daf695bac.png)

**Dynamic sequence length:**
`trtexec --loadEngine=model_default.plan --shapes=input_ids:1x7,attention_mask:1x7,token_type_ids:1x7`

![image](https://user-images.githubusercontent.com/37987800/186110528-63ce0e28-9e37-414c-81af-509b51715604.png)

**Note: Dynamic sequence length works fine while testing in tensorrt engine but it does not work when I deploy it on Triton server. Even if you look at my NLP model configuration file, BERT model input shapes are dynamic because dimension may change from one input = text/query to another.**

I also tried by passing my input text with shape of 1x128 then it works fine but it's not working for 1x7 or 1x<128. Please help me on this issue.

Thanks
